lib/chef/application/apply.rb

```diff
@@ -198,7 +208,11 @@ def run_chef_recipe
       @recipe_text, @recipe_fh = read_recipe_file @recipe_filename
     end
     recipe, run_context = get_recipe_and_run_context
-    recipe.instance_eval(@recipe_text, @recipe_filename, 1)
+    if config[:yaml]
```

Comment:
this should not really be in the config hash and should go into an instance variable at the very least.

i sort of feel like this should mostly be handled by Chef::Recipe rather than here, and if you pass a filename with .yml then Chef::Recipe should have an API which handles it.  but its complicated by the way that earlier in this method it optionally loads in a string off of STDIN or the cli.

you've also still got the filename here, and i'd almost prefer checking the extension of the filename twice rather than passing around state.

Reply:
right, this isn't just being used to avoid checking the filename a second time, this is for the case where you're passing `chef-apply --yaml`. I guess the above conditional with the logging could move down here too, but if we just did checked the file extension here then `something | chef-apply -s --yaml` wouldn't work.

I'd probably accept the argument that it shouldn't, because if you're doing that the training wheels should probably be off. But that's what I was doing.

Reply:
I removed this check on this line, but did keep logic here rather than Chef::Recipe to avoid changing the arguments to Chef::Recipe to add one forcing yaml mode there.

Reply:
Got it, i somehow missed the --yaml option.  but still not mutating that argument is probably better.